gapis/server/grpc.go

```diff
@@ -68,21 +62,40 @@ func NewWithListener(ctx context.Context, l net.Listener, cfg Config, srvChan ch
 			srvChan <- server
 		}
 		if cfg.IdleTimeout != 0 {
-			go s.stopIfIdle(ctx, server, keepAlive, cfg.IdleTimeout)
+			go s.stopIfIdle(ctx, server, cfg.IdleTimeout)
 		}
 		return nil
 	}, grpc.UnaryInterceptor(auth.ServerInterceptor(cfg.AuthToken)))
 }
 
 type grpcServer struct {
-	handler Server
-	bindCtx func(context.Context) context.Context
+	handler      Server
+	bindCtx      func(context.Context) context.Context
+	keepAlive    chan struct{}
+	inFlightRPCs uint32
+}
+
+// inRPC should be called at the start of an RPC call. The returned function
+// should be called when the RPC call finishes.
+func (s *grpcServer) inRPC() func() {
+	atomic.AddUint32(&s.inFlightRPCs, 1)
+	select {
+	case s.keepAlive <- struct{}{}:
+	default:
+	}
+	return func() {
+		select {
+		case s.keepAlive <- struct{}{}:
+		default:
+		}
+		atomic.AddUint32(&s.inFlightRPCs, ^uint32(0))
```

Comment:
Why use this xor "magic", when there is a signed version of the atomics? Are you really expecting 2147483648+ concurrent requests?